pangea-sdk/v3/service/store/api.go

```diff
@@ -0,0 +1,614 @@
+	Path           string                 `json:"path,omitempty"`
+	TransferMethod *pangea.TransferMethod `json:"transfer_method,omitempty"`
+}
+
+type GetResult struct {
+	Object  ItemData `json:"object"`
+	DestURL *string  `json:"dest_url,omitempty"`
+}
+
+func (e *store) Get(ctx context.Context, input *GetRequest) (*pangea.PangeaResponse[GetResult], error) {
+	return request.DoPost(ctx, e.Client, "v1beta/get", input, &GetResult{})
+}
+
+type PutRequest struct {
+	pangea.BaseRequest
+
+	Name           string                `json:"name,omitempty"`
+	Format         *FileFormat           `json:"format,omitempty"`
+	Metadata       Metadata              `json:"metadata,omitempty"`
+	MimeType       string                `json:"mimetype,omitempty"`
+	ParentID       string                `json:"parent_id,omitempty"`
+	Path           string                `json:"path,omitempty"`
+	CRC32C         string                `json:"crc32c,omitempty"`
+	MD5            string                `json:"md5,omitempty"`
+	SHA1           string                `json:"sha1,omitempty"`
+	SHA256         string                `json:"sha256,omitempty"`
+	SHA512         string                `json:"sha512,omitempty"`
+	Size           *int                  `json:"size,omitempty"`
+	Tags           Tags                  `json:"tags,omitempty"`
+	TransferMethod pangea.TransferMethod `json:"transfer_method,omitempty"`
+}
+
+type PutResult struct {
+	Object ItemData `json:"object"`
+}
+
+func (e *store) Put(ctx context.Context, input *PutRequest, file *os.File) (*pangea.PangeaResponse[PutResult], error) {
+	if input == nil {
+		return nil, errors.New("nil input")
+	}
+
+	if input.TransferMethod == pangea.TMpostURL {
+		var err error
+		params, err := pangea.GetUploadFileParams(file)
+		if err != nil {
+			return nil, err
+		}
+		input.CRC32C = params.CRC
+		input.SHA256 = params.SHA256
+		input.Size = pangea.Int(params.Size)
+	}
+
+	name := "file"
+	if input.TransferMethod == pangea.TMmultipart {
+		name = "upload"
+	}
+
+	fd := pangea.FileData{
+		File: file,
+		Name: name,
+	}
+
+	return request.DoPostWithFile(ctx, e.Client, "v1beta/put", input, &PutResult{}, fd)
+}
+
+type UpdateRequest struct {
+	pangea.BaseRequest
+
+	ID             string   `json:"id"`
+	Path           string   `json:"path,omitempty"`
+	AddMetadata    Metadata `json:"add_metadata,omitempty"`
+	RemoveMetadata Metadata `json:"remove_metadata,omitempty"`
+	Metadata       Metadata `json:"metadata,omitempty"`
+	AddTags        Tags     `json:"add_tags,omitempty"`
+	RemoveTags     Tags     `json:"remove_tags,omitempty"`
+	Tags           Tags     `json:"tags,omitempty"`
+	ParentID       string   `json:"parent_id,omitempty"`
+	UpdatedAt      string   `json:"updated_at,omitempty"`
+}
+
+type UpdateResult struct {
+	Object ItemData `json:"object"`
+}
+
+func (e *store) Update(ctx context.Context, input *UpdateRequest) (*pangea.PangeaResponse[UpdateResult], error) {
```

Comment:
I think Update is missing a parameter for the updated file?

![Screenshot 2024-01-09 at 1 35 48 PM](https://github.com/pangeacyber/pangea-go/assets/10459856/ab0ae578-bf27-4771-b923-90058787d321)

Reply:
I think it's not supported yet, checking with Jason...

Reply:
No, it's not.

Reply:
Thanks for checking. I'll go ahead and push a commit to update the doc string here.